FreeRTOS/Test/CMock/Makefile

```diff
@@ -5,99 +5,121 @@ include makefile.in
+    @echo -e 'run_col_formatted : same as formatted but will show the results in colors'
+    @echo -e 'coverage          : will run code coverage and generate html docs in $(BUILD_DIR)/coverage/index.html'
+    @echo -e 'all               : will build documentations and coverage, which builds and runs all tests'
+
+$(LIB_DIR)/libcmock.so : $(CMOCK_SRC_DIR)/cmock.c                              \
+                         $(CMOCK_SRC_DIR)/cmock.h                              \
+                         $(LIB_DIR)/libunity.so                                \
+                         Makefile | directories
+    $(CC) -o $@ -shared -fPIC $<  $(INCLUDE_DIR)
+
+
+$(LIB_DIR)/libunity.so : $(UNITY_SRC_DIR)/unity.c                              \
+                         $(UNITY_SRC_DIR)/unity.h                              \
+                         Makefile | directories
+    ${CC} -o $@ -shared -fPIC  $<
+
+$(LIB_DIR)/libunitymemory.so: ${UNITY_MEM_DIR}/unity_memory.c                  \
+                              ${UNITY_MEM_DIR}/unity_memory.h                  \
+                              ${LIB_DIR}/libunity.so                           \
+                              Makefile | directories
     ${CC} -o $@ -shared -fPIC $< ${INCLUDE_DIR}
 
-run : $(UNITS)
-    mkdir -p $(BUILD_DIR)
-    rm -f $(BUILD_DIR)/unit_test_report.txt
-    for f in $(BIN_DIR)/*; do                               \
-        $${f} | tee -a $(BUILD_DIR)/unit_test_report.txt;   \
-    done
-    cd $(BUILD_DIR) &&                              \
-        ruby $(UNITY_BIN_DIR)/parse_output.rb -xml  \
-        $(BUILD_DIR)/unit_test_report.txt
-
-run_col : $(UNITS) zero_coverage
+$(LIB_DIR)/libcexception.so: ${C_EXCEPTION_SRC_DIR}/CException.c               \
+                             ${C_EXCEPTION_SRC_DIR}/CException.h               \
+                             Makefile | directories
+   ${CC} -o $@ -shared -fPIC $< ${INCLUDE_DIR}
+
+run : $(UNITS) | directories
+    -rm -f $(BUILD_DIR)/unit_test_report.txt
+    for f in $(BIN_DIR)/*; do                                                  \
+        $${f} | tee -a $(BUILD_DIR)/unit_test_report.txt ;                     \
+    done 
+    cd $(BUILD_DIR) &&                                                         \
+    ruby $(UNITY_BIN_DIR)/parse_output.rb                                      \
+        -xml $(BUILD_DIR)/unit_test_report.txt
+
+run_col : $(UNITS) zero_coverage | directories
     for f in $(BIN_DIR)/*; do \
-        ruby -r $(UNITY_BIN_DIR)/colour_reporter.rb -e "report('`$${f}`')"; done
+        ruby -r $(UNITY_BIN_DIR)/colour_reporter.rb  -e "report('`$${f}`')";   \
+    done
 
-run_formatted : $(UNITS) zero_coverage
-    for f in $(BIN_DIR)/*; do                                       \
-        $${f} > $(BUILD_DIR)/output;                                \
-        ruby $(UNITY_BIN_DIR)/parse_output.rb $(BUILD_DIR)/output ; \
+run_formatted :  $(UNITS) zero_coverage | directories
```

Comment:
This "| directories" notation breaks down when you try to build a test by itself. It's easier to just call mkdir -p for each target directory.

Reply:
i am calling tests alone, it is working fine, maybe i fixed something with the new changes
thats what the | is normally used for

Reply:
Try running the following:
```
cd FreeRTOS/FreeRTOS/Test/CMock
make clean
make -C tasks
```

Reply:
$ make clean
```
shared libs are /home/gedeonag/workspace/FreeRTOS/FreeRTOS/Test/CMock/build/lib/libunity.so /home/gedeonag/workspace/FreeRTOS/FreeRTOS/Test/CMock/build/lib/libcmock.so /home/gedeonag/workspace/FreeRTOS/FreeRTOS/Test/CMock/build/lib/libunitymemory.so /home/gedeonag/workspace/FreeRTOS/FreeRTOS/Test/CMock/build/lib/libcexception.so
rm -rf /home/gedeonag/workspace/FreeRTOS/FreeRTOS/Test/CMock/build
```
$ make tasks
```
...
ad -lunity -lunitymemory -lcexception -lcmock -lgcov  -o /home/gedeonag/workspace/FreeRTOS/FreeRTOS/Test/CMock/build/bin/tasks_utest_2
make[1]: Leaving directory `/home/gedeonag/workspace/FreeRTOS/FreeRTOS/Test/CMock/tasks'
```

ehco $?
```
0
```